src/electron/platform/android/fetch-scan-results.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import axios from 'axios';
+import { ScanResults } from './scan-results';
+
+export function fetchScanResults(port: number): Promise<ScanResults> {
+    return new Promise<any>((resolve, reject) => {
```

Comment:
You can simplify this as following:

```ts
export function fetchScanResults(port: number): Promise<ScanResults> {
    return axios
        .get(`http://localhost:${port}/axe/result`)
        .then(response => {
            return new ScanResults(response.data);
        })
        .catch(error => {
            return error;
        });
}
```
It will still return a promise, you dont need to wrap this in a `new Promise` to return a `Promise`.
When you call `fetchScanResults`; at that moment, you can handle the error properly.

Reply:
You can also leave out the `catch` part here and just let it propagate to the  `checkPort` function; you are already handling the `Promise` error in that function.

Reply:
The suggested code (with the catch) caused the returned promise's reject method to not get called correctly. It worked if I removed the catch completely, but then it's dependent upon the existence of an upstream catch handler, which we can't guarantee. To retain the error handling without making assumptions about the upstream code (remembering that `checkPort` is just for testing), I'm going to stick with returning a `new Promise`